src/brsTypes/components/RoArray.ts

```diff
@@ -266,48 +310,31 @@ export class RoArray extends BrsComponent implements BrsValue, BrsIterable {
             if (flags.toString().match(/([^ir])/g) != null) {
                 interpreter.stderr.write("roArray.SortBy: Flags contains invalid option(s).\n");
             } else {
-                this.elements = this.elements.sort(function (a, b) {
+                const caseInsensitive = flags.toString().indexOf("i") > -1;
+                const originalArrayCopy = [...this.elements];
+                this.elements = this.elements.sort((a, b) => {
                     var compare = 0;
                     if (a instanceof RoAssociativeArray && b instanceof RoAssociativeArray) {
-                        if (
-                            a.elements.has(fieldName.toString().toLowerCase()) &&
-                            b.elements.has(fieldName.toString().toLowerCase())
-                        ) {
+                        let aHasField = a.elements.has(fieldName.toString().toLowerCase()),
+                            bHasField = b.elements.has(fieldName.toString().toLowerCase());
```

Comment:
I don't think there's a lint rule around this (I'm sorry!) but local convention is to use separate `let` or `const` keywords for each declaration, like you did on lines 313 and 314.  Would you mind adjusting these?

```suggestion
                        let aHasField = a.elements.has(fieldName.toString().toLowerCase());
                        let bHasField = b.elements.has(fieldName.toString().toLowerCase());
```